INSTALL.md

```diff
@@ -1,143 +1,125 @@
-* `[develop]` Specify optional dependency groups. You can omit any or all of these if your system does not support them or if you do not need them.
+**Note** that you can work with your installation using the usual git commands (pull, push, etc.). However, these ***won't recompile*** any of the extension (i.e. pyx, _C/C++_) libraries. See below for how to reinstall PISA when you need these to recompile.
 
-### **Test your installation by running a simple script:**
 
-- Activate your python environment (`pisa_env`)
+### Reinstall PISA
 
-  ```bash
-  conda activate pisa_env
-  ```
-- Start python interpretator in terminal
+Sometimes a change within PISA requires re-installation (particularly if a compiled module changes, the below forces re-compilation).
+
+```bash
+pip install -e $PISA[develop] --force-reinstall -vvv
+```
+
+**Note** that if files change names or locations, though, the above can still not be enough.
+In this case, the old files have to be removed manually (along with any associated `.pyc` files, as Python will use these even if the `.py` files have been removed).
+them.
+
+
+### Test PISA
+
+ First activate your python environment (if not already active). Assuming you used Miniconda, you can do that with:
 
   ```bash
-  python
+  source PATH_TO_ANACONDA/miniconda3/bin/activate
+  conda activate NAME_OF_YOUR_PISA_ENV
   ```
-- Import `Pipline` modelue and `matplotlib.pyplot`
 
-  ```python
-  from pisa.core import Pipeline
-  import matplotlib.pyplot as plt
-  ```
-  ```
-  << PISA is running in single precision (FP32) mode; numba is running on CPU (single core) >>
-  ```
-- Load an example pipeline
+#### Running a simple script
 
-  ```python
-  template_maker = Pipeline("settings/pipeline/osc_example.cfg")
-  ```
-- Run the pipeline
+Running the following command should create a folder in `/tmp` containing a pdf with output maps for different neutrino types and interactions.
 
-  ```python
-  template_maker.run()
-  ```
-- Get the oscillation probabilities as a map
+```bash
+$PISA/pisa/core/distribution_maker.py --pipeline settings/pipeline/IceCube_3y_neutrinos.cfg --outdir /tmp/pipeline_output --pdf
+```
 
-  ```python
-  outputs = template_maker.data.get_mapset('prob_mu')
-  ```
-- Plot the oscillogram
+#### Unit Tests
 
-  ```python
-  fig, axes = plt.subplots(figsize=(18, 5), ncols=3)
-  outputs['nue_cc'].plot(ax=axes[0], cmap='RdYlBu_r', vmin=0, vmax=1);
-  outputs['numu_cc'].plot(ax=axes[1], cmap='RdYlBu_r', vmin=0, vmax=1);
-  outputs['nutau_cc'].plot(ax=axes[2], cmap='RdYlBu_r', vmin=0, vmax=1);
-  ```
+Throughout the codebase there are `test_*.py` files and `test_*` functions within various `*.py` files that represent unit tests.
+Unit tests are designed to ensure that the basic mechanisms of objects' functionality work.
+
+These are all run, plus additional tests (takes about 15-20 minutes on a laptop) with the command
+```bash
+$PISA/pisa_tests/test_command_lines.sh
```

Comment:
Have you looked at `test_command_lines.sh`? Many of the scripts it is trying to run don't exist or are broken. I would suggest running something else instead.